runtime/gc/gc.go

```diff
@@ -89,3 +169,31 @@ func (gc *GarbageCollector) CollectAll() error {
 	gc.resources = resources
 	return nil
 }
+
+// needDiskSpace returns true if we need to free diskspace
+func (gc *GarbageCollector) needDiskSpace() bool {
+	// If we have no metrics or minimum diskspace we remove everything
+	if gc.minimumDiskSpace == 0 || gc.storageFolder == "" {
+		return true
+	}
+	stat, err := disk.Usage(gc.storageFolder)
+	if err != nil {
```

Comment:
hmm.. yeah probably, we don't have a log context, I'll leave a todo and do it later... I have too many PRs built ontop of this to do major refactoring..